streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsRebalanceListener.java

```diff
@@ -60,6 +60,9 @@ public void onPartitionsAssigned(final Collection<TopicPartition> partitions) {
         }  else if (assignmentErrorCode.get() == AssignorError.ASSIGNMENT_ERROR.code()) {
             log.error("Received error code {}", AssignorError.ASSIGNMENT_ERROR);
             throw new TaskAssignmentException("Hit an unexpected exception during task assignment phase of rebalance");
+        } else if (assignmentErrorCode.get() == AssignorError.SHUTDOWN_REQUESTED.code()) {
+            log.error("An application is requesting Shutdown");
```

Comment:
```suggestion
            log.error("A Kafka Streams client in this Kafka Streams application is requesting to shutdown");
```
An application is actually a group of Kafka Streams clients (or instances).